code/game/machinery/teleporter.dm

```diff
@@ -297,6 +298,14 @@
 	if (!com.target)
 		visible_message("<span class='alert'>Cannot authenticate locked on coordinates. Please reinstate coordinate matrix.</span>")
 		return
+	if(com.target.z != z && z != ZLEVEL_STATION) //if our target is a different zlevel and our zlevel isn't the station
+		if(blockExiles(M))
+			if(!com.spam_flag)
+				com.spam_flag = TRUE
+				com.say("Exile implant detected. Teleportation denied.")
+				spawn(30)
```

Comment:
i tried using addtimer here but that garbage wouldn't work after at least an hour of troubleshooting so i ended up getting rid of the message completely